milabench/common.py

```diff
@@ -296,6 +308,160 @@ def _read_reports(*runs):
     return all_data
 
 
+def _find_metas(reports):
+    local_meta = next(iter(e for _r in reports for e in _r if e["event"] == "meta"), None)
+    if local_meta:
+        local_meta = local_meta["data"]
+    remote_metas = []
+    for _r in reports:
+        meta_lines = []
+        for event in _r:
+            _, event_type, line = None, "", []
+
+            try:
+                _, event_type, *line = event["data"].split(" ")
+            except (AttributeError, ValueError):
+                pass
+
+            if event_type[:1] + event_type[-1:] != "[]":
+                event_type = None
+                line = event["data"]
+            else:
+                line = " ".join(line)
+
+            if event_type == "[meta]":
+                meta_lines.append(line)
+            elif event_type is None and meta_lines:
+                meta_lines.append(line)
+            elif meta_lines:
+                remote_metas.append(yaml.safe_load("".join(meta_lines)))
+                meta_lines = []
+
+    return local_meta, remote_metas
+
+
+def _filter_reports(*reports):
+    all_reports = []
+
+    for report in reports:
+        config = next(iter(e for e in report if e["event"] == "config"), None)
+        if config is None:
+            continue
+
+        if config["data"]["name"] != "remote":
+            all_reports.append(report)
+
+    return all_reports
+
+
+def _push_reports(reports_repo, runs, packs:dict=None):
+    _SVG_COLORS = {
+        "pass": "blue",
+        "partial": "yellow",
+        "failure": "red",
+    }
+    import milabench.cli.badges as badges
+
+    _repo = git.repo.base.Repo(ROOT_FOLDER)
+    try:
+        reports_repo = git.repo.base.Repo(str(reports_repo))
+    except (git.exc.InvalidGitRepositoryError, git.exc.NoSuchPathError):
+        repo_url = next(iter(_r.url for _r in _repo.remotes if _r.name == "origin"), None)
+        reports_repo = git.repo.base.Repo.clone_from(repo_url, str(reports_repo), branch="reports")
+
+    reports_url = ([
+        _r.url for _r in _repo.remotes if "mila-iqia" in _r.url
+    ] or [
+        _r.url for _r in _repo.remotes if _r.name == "origin"
+    ])[0]
+    reports_url = XPath("github.com".join(reports_url.split("github.com")[1:])[1:])
+    reports_url = XPath("https://github.com") / f"{reports_url.with_suffix('')}/tree/{reports_repo.active_branch.name}"
```

Comment:
This might not be needed if we set the link with markdown instead of the svg itself to redirect to the summary page